yucca/training/data_loading/YuccaDataset.py

```diff
@@ -116,6 +123,44 @@ def _transform(self, data_dict, case):
             data_dict = self.composed_transforms(data_dict)
         return self.to_torch(data_dict)
 
+    def replace_missing_modalities_with_zero_arrays(self, data):
+        assert data.dtype == "object", "allow missing modalities is true but dtype is not object"
+        # First find the array with the largest array.
+        # in classification this avoids setting the zero array to the 1d array with classes
+        sizes = [i.size for i in data]
+        idx_largest_array = np.where(sizes == np.max(sizes))[0][0]
+        for idx, i in enumerate(data):
+            if i.size == 0:
+                data[idx] = np.zeros(data[idx_largest_array].squeeze().shape)
+        return data
+
+    def unpack(self, data):
+        if self.supervised:
```

Comment:
As a rule of thumb, i think the readability is impacted negatively when a method accesses data which is not provided directly as an argument. I would propose providing `self.supervised` with a type annotation as an argument and only use the properties when it is absolutely necessary.